pkg/kubelet/lifecycle/predicate.go

```diff
@@ -127,9 +134,17 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
 		r := reasons[0]
 		switch re := r.(type) {
 		case *predicates.PredicateFailureError:
-			reason = re.PredicateName
-			message = re.Error()
-			klog.V(2).Infof("Predicate failed on Pod: %v, for reason: %v", format.Pod(admitPod), message)
+			// only tolerant MatchNodeSelector predicate failure reason
+			if tolerant && re.PredicateName == predicates.MatchNodeSelectorPred {
```

Comment:
special casing this one selector seems fragile... it would be reasonable for other things in this method would assume they had usable nodeInfo available to them